exercises/complex-numbers/description.md

```diff
@@ -1,29 +1,95 @@
 # Description
 
-A complex number is a number in the form `a + b * i` where `a` and `b` are real and `i` satisfies `i^2 = -1`.
+A **complex number** is expressed in the form `z = a + b * i`, where:
 
-`a` is called the real part and `b` is called the imaginary part of `z`.
-The conjugate of the number `a + b * i` is the number `a - b * i`.
-The absolute value of a complex number `z = a + b * i` is a real number `|z| = sqrt(a^2 + b^2)`. The square of the absolute value `|z|^2` is the result of multiplication of `z` by its complex conjugate.
+- `a` is the **real part** (a real number),
 
-The sum/difference of two complex numbers involves adding/subtracting their real and imaginary parts separately:
-`(a + i * b) + (c + i * d) = (a + c) + (b + d) * i`,
-`(a + i * b) - (c + i * d) = (a - c) + (b - d) * i`.
+- `b` is the **imaginary part** (also a real number), and
 
-Multiplication result is by definition
-`(a + i * b) * (c + i * d) = (a * c - b * d) + (b * c + a * d) * i`.
+- `i` is the **imaginary unit** satisfying `i^2 = -1`.
 
-The reciprocal of a non-zero complex number is
-`1 / (a + i * b) = a/(a^2 + b^2) - b/(a^2 + b^2) * i`.
+## Key Properties
 
-Dividing a complex number `a + i * b` by another `c + i * d` gives:
-`(a + i * b) / (c + i * d) = (a * c + b * d)/(c^2 + d^2) + (b * c - a * d)/(c^2 + d^2) * i`.
+### Conjugate
 
-Raising e to a complex exponent can be expressed as `e^(a + i * b) = e^a * e^(i * b)`, the last term of which is given by Euler's formula `e^(i * b) = cos(b) + i * sin(b)`.
+The conjugate of the complex number `z = a + b * i` is given by:
 
-Implement the following operations:
+```math
+z̅ = a - b * i
+```
 
-- addition, subtraction, multiplication and division of two complex numbers,
-- conjugate, absolute value, exponent of a given complex number.
+### Absolute Value
 
-Assume the programming language you are using does not have an implementation of complex numbers.
+The absolute value (or modulus) of `z` is defined as:
+
+```math
+|z| = sqrt(a^2 + b^2)
+```
+
+The square of the absolute value, `|z|²`, can be computed as the product of `z` and its conjugate:
```

Comment:
Square isn't part of the tests. Should it be here? If yes, should it have a header?

Reply:
Thank you for pointing this out! Since the square of the absolute value isn't part of the tests or required to solve the exercise, I think it might be better to leave it out altogether. Including it could potentially confuse learners by making them think it's something they need to implement.

On a related note, I noticed that the reciprocal isn't tested either. I included it because it was in the previous version of the file, but if you feel it’s unnecessary, I can remove that as well.

Let me know your thoughts, and I’ll make the changes accordingly. 😊

Reply:
The square of a complex number could be used in a division operation (the square of the second number is the denominator in both fractions), so it might be preferable to leave it for those who wish to explore that approach:

```(a + i * b) / (c + i * d) = (a * c + b * d)/(c^2 + d^2) + (b * c - a * d)/(c^2 + d^2) * i```

The reciprocal section doesn't seem to be useful at all and can likely be safely removed.

Reply:
You're right that the reciprocal doesn't have a direct use in the given context. It might be the case that the author included it initially as a way to offer an alternative approach or to keep the option open for students who want to explore multiplying by the reciprocal, as it essentially performs division. But in terms of the immediate solution, it does seem unnecessary.

Reply:
True, dividing by a number is the same as multiplying by its reciprocal, just like with real numbers. However, this might not be immediately obvious, especially when working with complex numbers. Looking through community solutions, I noticed that some people use the square method, while none seem to use the reciprocal.

Also, I noticed that in the new instructions, the reciprocal is introduced after the division. This might cause people to focus more on the division formula and ignore the reciprocal as an option. In the original instructions, the reciprocal was mentioned right after multiplication, though it didn't seem to make any difference overall. If we still want people to consider the reciprocal route, we might want to explicitly indicate its connection with the division operation, like so:

`z₁ / z₂ = z₁ * (1 / z₂) = <formula goes here>`

I don't have a strong opinion on this, but since I'm biased due to my background, I'll stay out of the decision. Some might argue that the instructions are too technical and prefer removing both the square and the reciprocal altogether.

Reply:
I agree that removing the reciprocal concept could simplify things and make the instructions more straightforward, especially for those who might find it confusing when dealing with complex numbers. Simplifying it would likely make the learning process smoother, so I think it could be a great move. Thanks for sharing your thoughts!

Reply:
Keep in mind that there are already 5 approvals, so it might be better to leave the discussion on whether to remove the square and reciprocal for another thread. As it stands, I would still suggest moving the reciprocal before the division (as it was in the original instructions) and just adding the `z₁ * (1 / z₂)` to give everyone the chance to use the reciprocal if they want.

Reply:
Sure, I've changed it accordingly!

Reply:
1. There's a typo in the header.
2. I think the Square section ought to be dropped. It's not relevant to this exercise. Reciprocal is useful for division, so it can be useful and bumped to a later discussion. I'm not sure why Square would be kept.

Reply:
Upon reflecting further, I realized that the header isn't necessary at all, as the property in question is the square of the absolute value of the complex number, not the complex number itself. Regarding its significance, this property is widely used in various operations, from division to finding the reciprocal of a complex number. If you examine the formulas, you'll notice it appears in the denominators of both the real and imaginary components.

Reply:
That's correct, all denominators are the square of either the first or the complex number.

Here's a solution that defines it on line 7 and uses it for division on line 29

https://exercism.org/tracks/clojure/exercises/complex-numbers/solutions/fergalbyrne